src/modules/transaction/txUtils.ts

```diff
@@ -9,31 +13,30 @@ import {
   RevertedTransaction,
   ConfirmedTransaction
 } from './types'
-import { createEth } from '../../lib/eth'
 
 export async function getTransaction(
+  address: string,
+  chainId: ChainId,
   hash: string
 ): Promise<AnyTransaction | null> {
-  const eth = await createEth()
-  if (!eth) return null
+  const provider = await connection.createProvider(
```

Comment:
Should we need to use `getProvider(chainId)` here?

Reply:
I wanted to be more specific and wasn't really sure about the new signature of `getProvider`. The new proposed implementation of `getConnectedProvider` makes this a lot more reliable. I'll change it :)